hbase-server/src/test/java/org/apache/hadoop/hbase/mob/TestMobCompactionBase.java

```diff
@@ -0,0 +1,242 @@
+        r.nextBytes(key);
+        Put p = new Put(key);
+        p.addColumn(fam, qualifier, mobVal);
+        table.put(p);
+      }
+      admin.flush(table.getName());
+      LOG.info("Finished loading {} rows", num);
+    } catch (Exception e) {
+      LOG.error("MOB file compaction chore test FAILED", e);
+      assertTrue(false);
+    }
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    HTU.shutdownMiniCluster();
+  }
+
+
+  public void baseTestMobFileCompaction() throws InterruptedException, IOException {
+
+    try {
+
+      // Load and flush data 3 times
+      loadData(rows);
+      loadData(rows);
+      loadData(rows);
+      long num = getNumberOfMobFiles(conf, table.getName(), new String(fam));
+      assertEquals(numRegions * 3, num);
+      // Major MOB compact
+      mobCompact(admin, hdt, hcd);
+      // wait until compaction is complete
+      while (admin.getCompactionState(hdt.getTableName()) != CompactionState.NONE) {
+        Thread.sleep(100);
+      }
+
+      num = getNumberOfMobFiles(conf, table.getName(), new String(fam));
+      assertEquals(numRegions * 4, num);
+      // We have guarantee, that compcated file discharger will run during this pause
+      // because it has interval less than this wait time
+      LOG.info("Waiting for {}ms", minAgeToArchive + 1000);
+
+      Thread.sleep(minAgeToArchive + 1000);
+      LOG.info("Cleaning up MOB files");
+      // Cleanup again
+      cleanerChore.cleanupObsoleteMobFiles(conf, table.getName());
+
+      num = getNumberOfMobFiles(conf, table.getName(), new String(fam));
+      assertEquals(numRegions, num);
+
+      long scanned = scanTable();
+      assertEquals(3 * rows, scanned);
+
+    } finally {
+
+      admin.disableTable(hdt.getTableName());
+      admin.deleteTable(hdt.getTableName());
+    }
+
+  }
+
+  protected abstract void mobCompact(Admin admin2, HTableDescriptor hdt2, HColumnDescriptor hcd2)
+      throws IOException, InterruptedException;
+
+
+  protected  long getNumberOfMobFiles(Configuration conf, TableName tableName, String family)
+      throws IOException {
+    FileSystem fs = FileSystem.get(conf);
+    Path dir = MobUtils.getMobFamilyPath(conf, tableName, family);
+    FileStatus[] stat = fs.listStatus(dir);
+    for (FileStatus st : stat) {
+      LOG.debug("MOB Directory content: {}", st.getPath());
+    }
+    LOG.debug("MOB Directory content total files: {}", stat.length);
+
+    return stat.length;
+  }
+
+
+  protected long scanTable() {
+    try {
+
+      Result result;
+      ResultScanner scanner = table.getScanner(fam);
+      long counter = 0;
+      while ((result = scanner.next()) != null) {
+        assertTrue(Arrays.equals(result.getValue(fam, qualifier), mobVal));
+        counter++;
+      }
+      return counter;
+    } catch (Exception e) {
+      e.printStackTrace();
+      LOG.error("MOB file compaction test FAILED");
+      if (HTU != null) {
+        assertTrue(false);
```

Comment:
use `fail` and include a message about why we're failing so junit can propagate it.